inc/dashboard/namespace.php

```diff
@@ -0,0 +1,421 @@
+	}
+
+	return $data;
+}
+
+/**
+ * Get analytics views list data.
+ *
+ * @param int $days How many days worth of analytics to fetch. Defaults to 7.
+ *
+ * @return array The array of list data from ElasticSearch.
+ */
+function get_views_list( int $days = 7 ) : array {
+	$date_start = time() - ( $days * DAY_IN_SECONDS );
+
+	$query = [
+		'query' => [
+			'bool' => [
+				'filter' => [
+					[
+						// Query from the current site.
+						'term' => [
+							'attributes.blogId.keyword' => get_current_blog_id(),
+						],
+					],
+					[
+						// We're interested in views and conversions.
+						'terms' => [
+							'event_type.keyword' => [
+								'experienceView',
+								'conversion',
+							],
+						],
+					],
+					[
+						// Limit to date range.
+						'range' => [
+							'event_timestamp' => [
+								'gte' => $date_start * 1000,
+							],
+						],
+					],
+				],
+			],
+		],
+		'aggs' => [
+			'blocks' => [
+				'terms' => [
+					// Get the block data. This will give us the key for the block, which is stored as the post slug.
+					'field' => 'attributes.clientId.keyword',
+					'size' => 10000, // Use arbitrary large size that is more than we're likely to need.
+				],
+				'aggs' => [
+					'views' => [
+						'filter' => [
+							'term' => [
+								'event_type.keyword' => 'experienceView',
+							],
+						],
+						'aggs' => [
+							'uniques' => [
+								'cardinality' => [
+									'field' => 'endpoint.Id.keyword',
+								],
+							],
+						],
+					],
+					'conversions' => [
+						'filter' => [
+							'term' => [
+								'event_type.keyword' => 'conversion',
+							],
+						],
+						'aggs' => [
+							'uniques' => [
+								'cardinality' => [
+									'field' => 'endpoint.Id.keyword',
+								],
+							],
+						],
+					],
+					'conversion_rate' => [
+						'bucket_script' => [
+							'buckets_path' => [
+								'views' => 'views>uniques',
+								'conversions' => 'conversions>uniques',
+							],
+							'script' => 'params.conversions / params.views',
+						],
+					],
+				],
+			],
+		],
+		'size' => 0,
+		'sort' => [
+			'event_timestamp' => 'desc',
+		],
+	];
+
+	$key = sprintf( 'views:list:days:%d', $days );
+	$cache = wp_cache_get( $key, 'altis-xbs' );
+
+	if ( $cache ) {
+		return $cache;
+	}
+
+	$result = Utils\query( $query );
+
+	if ( ! $result ) {
+		$data = [];
+
+		wp_cache_set( $key, $data, 'altis-xbs', MINUTE_IN_SECONDS );
+		return $data;
+	}
+
+	$data = get_aggregate_data( $result['aggregations']['blocks']['buckets'] ?? [] );
+	wp_cache_set( $key, $data, 'altis-xbs', 5 * MINUTE_IN_SECONDS );
+
+	return $data;
+}
+
+/**
+ * Alter the default WP_Query to change the sort order.
+ *
+ * @param \WP_Query $query The WP_Query object.
+ *
+ * @return \WP_Query The updated WP_Query object.
+ */
+function modify_views_list_query( $query ) {
+	// Bail if we aren't looking at the XB Insights page.
+	if ( $query->get( 'post_type' ) !== Blocks\POST_TYPE ) {
+		return $query;
```

Comment:
This is an action rather than a filter so there's no need to return any value, we're modifying the reference to the query object.
```suggestion
		return;
```